pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -158,6 +163,25 @@ func (c *controller) getSecretData(machineClassName string, secretRefs ...*v1.Se
 	return secretData, nil
 }
 
+// validateNodeTemplate validates if the capacity field has CPU, GPU and memory configured when
+// the optional nodeTemplate field is configured in the MachineClass
+func (c *controller) validateNodeTemplate(nodeTemplate *v1alpha1.NodeTemplate) error {
+	capacityAttributes := []v1.ResourceName{"cpu", "gpu", "memory"}
+
+	if nodeTemplate == nil {
+		return nil
+	}
+
+	for _, attribute := range capacityAttributes {
+		if _, ok := nodeTemplate.Capacity[attribute]; !ok {
+			err := errors.New("MachineClass NodeTemplate Capacity should mandatorily have CPU, GPU and Memory configured")
+			return err
+		}
+	}
+
```

Comment:
we can add validation for instanceType , region and zone , validating that empty string is not provided

Reply:
the docstring of the method will have to be adopted accordingly

Reply:
good point too.